dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -877,6 +881,34 @@ public void download(final List<String> args) {
         }
     }
 
+    /**
+     * Validates that any JSON and/or YAML files are syntactically valid.
+     *
+     * @param args
+     */
+    protected void preValidateLaunchArguments(List<String> args) {
+        if (args.contains("--json")) {
+            String jsonFile = reqVal(args, "--json");
+            try {
+                fileToJSON(jsonFile);
+            } catch (ParserException ex) {
+                errorMessage("Could not launch, syntax error in json file: " + jsonFile, CLIENT_ERROR);
+            } catch (IOException e) {
+                errorMessage("Could not launch, syntax error in json file: " + jsonFile, CLIENT_ERROR);
```

Comment:
You don't want the syntax error if you get an IOException. You would get an IOException, for example, if the file didn't exist, you don't have permissions to read it, etc.

You could just ignore non-`ParserException` errors and let the current error handling handle those cases. But then this makes this code sort of odd. Not sure what the answer is.

Reply:
In either case, the message should be different/might be able to extract something useful from the IOException message.